src/cli/cli.cc

```diff
@@ -88,64 +103,45 @@ static StatusCode pullAndInstall(AkliteClient &client, int version, const std::s
-  CheckInResult cr{CheckInResult::Status::Failed, "", std::vector<TufTarget>{}};
-  if (local_update_source == nullptr) {
-    cr = client.CheckIn();
-  } else {
-    cr = client.CheckInLocal(local_update_source);
-  }
-  if (!cr) {
-    return res2StatusCode<CheckInResult::Status>(c2s, cr.status);
-  }
+  auto cr = client.GetTargetToInstall(local_update_source, version, target_name, true, true);
 
-  auto target = cr.SelectTarget(version, target_name);
-  if (target.IsUnknown()) {
-    std::string target_string;
-    if (version == -1 && target_name.empty()) {
-      target_string = " version: latest,";
-    } else {
-      if (!target_name.empty()) {
-        target_string = " name: " + target_name + ",";
-      }
-      if (version != -1) {
-        target_string += " version: " + std::to_string(version) + ",";
+  //
+  if (cr.selected_target.IsUnknown()) {
+    if (cr.status == GetTargetToInstallResult::Status::NoMatchingTargets) {
+      std::string target_string;
+      if (version == -1 && target_name.empty()) {
+        target_string = " version: latest,";
+      } else {
+        if (!target_name.empty()) {
+          target_string = " name: " + target_name + ",";
+        }
+        if (version != -1) {
+          target_string += " version: " + std::to_string(version) + ",";
+        }
       }
-    }
 
-    LOG_ERROR << "No Target found;" << target_string
-              << " hardware ID: " << client.GetConfig().get("provision.primary_ecu_hardware_id", "")
-              << ", tag: " << client.GetConfig().get("pacman.tags", "");
-    return StatusCode::TufTargetNotFound;
+      LOG_ERROR << "No Target found;" << target_string
+                << " hardware ID: " << client.GetConfig().get("provision.primary_ecu_hardware_id", "")
+                << ", tag: " << client.GetConfig().get("pacman.tags", "");
+    } else if (cr) {
+      LOG_INFO << "No target to update";
+    }
+    return res2StatusCode<GetTargetToInstallResult::Status>(t2s, cr.status);
   }
 
-  if (current.Version() > target.Version()) {
+  const auto current{client.GetCurrent()};
+  if (current.Version() > cr.selected_target.Version()) {
     LOG_WARNING << "Found TUF Target is lower version than the current on; "
-                << "current: " << current.Version() << ", found Target: " << target.Version();
+                << "current: " << current.Version() << ", found Target: " << cr.selected_target.Version();
 
     if (!force_downgrade) {
       LOG_ERROR << "Downgrade is not allowed by default, re-run the command with `--force` option to force downgrade";
       return StatusCode::InstallDowngradeAttempt;
     }
-    LOG_WARNING << "Downgrading from " << current.Version() << " to  " << target.Version() << "...";
-  }
-
-  // Check whether the given target is already installed and synced/running
-  if (current == target && client.CheckAppsInSync() == nullptr) {
-    if (local_update_source != nullptr) {
-      return StatusCode::InstallAlreadyInstalled;
```

Comment:
It looks like we lost `InstallAlreadyInstalled` case?

Reply:
I've re-added it now